src/Stream.ts

```diff
@@ -52,7 +52,7 @@ export const switchMap = <E, X, E2>(fx: Fx.Fx<E, X>, f: (a: Event<E>) => Fx.Fx<E
 
       return x
     })
-  )
+  ) as Fx.Fx<ExcludeStream<E, Fork.Fork | Async.Async | E2>, X>
```

Comment:
It's a bummer that this requires an assertion now.  I fiddled with it a lot, but couldn't find a formulation of ExcludeStream that worked without an assertion.  I may try removing the ExcludeStream type and using Fx.Handle directly, to see if that helps.

Reply:
It might be related to `new CurrentTask<ExcludeStream<E> | E2>()`.
I think it should be something like `new CurrentTask<ExcludeStream<E,  E2>>()` so the conditional types are in alignment

Reply:
Good thought.  I tried, but still ended up needing the type assertion.  In the process, I realized it can be just `CurrentTask<E2>`, but even then the type assertion is needed.

Reply:
Yeah I attempted it locally too, and it doesn't seem to agree with the conditional types from `CurrentTask.run`. The type-cast probably is necessary for now